### PR TITLE
[stable/jenkins] Master.InitContainerEnv: Init Container Env Vars

### PR DESCRIPTION
**Pull request by StuartHarris**
- [x] Add Master.InitContainerEnv to allow sending env vars into init container (e.g. `http_proxy`)
- [x] Add Master.ContainerEnv to allow sending env vars into the main Jenkins container
- [x] Add section to Readme to potentially help those running behind a corporate proxy

Note: there are also some cosmetic changes in the readme due to formatting by `prettier`, e.g. table formatting where the only real change is the addition of `Master.InitContainerEnv`. It may be easier to see the real changes in the readme by ignoring whitespace when viewing the diff: https://github.com/kubernetes/charts/pull/3495/files&w=1

### CODE REVIEW

File: stable/jenkins/values.yaml [hunk 1]
```diff
@@ -15,6 +15,14 @@ Master:
   # AdminPassword: <defaults to random>
   Cpu: "200m"
   Memory: "256Mi"
+  # Environment variables that get added to the init container (useful for e.g. http_proxy)
+  # InitContainerEnv:
```
Comment (samwhite) [inline stable/jenkins/values.yaml:24]: I think we only need one example env var here in the comments - let's keep `http_proxy` but remove the other two? We can keep all three in the readme for completeness

**Review by samwhite — APPROVED** [review]
Useful change, makes it much easier to get Jenkins started behind a corporate proxy 👍I suppose an alternative approach would be to move anything that requires access to the Internet out of the InitContainer (e.g. downloading plugins) and moving them in to the main `jenkins` container somehow- as it's much easier to put the HTTP proxy setup in there through the `-D` flags

**Comment by sameersbn** [comment]
This is awesome. How do you feel about having similar options for the containers in the deployment?

**Comment by StuartHarris** [comment]
Thanks. Yeah, I think that's a good idea. I'll add the same again for the main container.

File: stable/jenkins/values.yaml (hunk 1 above)
Comment (StuartHarris) [reply stable/jenkins/values.yaml:24]: Good point, thanks. Will just leave one example in the comments.
Regarding the Init Container, I think it's good practice to have all the initial setup in an Init Container, so the main container can be as vanilla as possible.

**Comment by StuartHarris** [comment]
@sameersbn Hi Sameer, I added env vars for the main container too. Thanks :-)

**Comment by StuartHarris** [comment]
@sameersbn please take a look when you get a chance - thanks.

**Comment by StuartHarris** [comment]
@unguiculus thanks, but typo :-)

**Pull request merged**